service/controller/v11/resource/pod/delete.go

```diff
@@ -2,8 +2,116 @@ package pod
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/reconciliationcanceledcontext"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/kvm-operator/service/controller/v11/key"
 )
 
 func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	reconciledPod, err := key.ToPod(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "looking for the current version of the reconciled pod in the Kubernetes API")
+
+	var currentPod *corev1.Pod
+	{
+		currentPod, err = r.k8sClient.CoreV1().Pods(reconciledPod.GetNamespace()).Get(reconciledPod.Name, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// In case we reconcile a pod we cannot find anymore this means the
+			// informer's watch event is outdated and the pod got already deleted in the
+			// Kubernetes API. This is a normal transition behaviour, so we just ignore
+			// it and assume we are done.
+			r.logger.LogCtx(ctx, "debug", "cannot find the current version of the reconciled pod in the Kubernetes API")
+			reconciliationcanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "debug", "canceling reconciliation for pod")
+
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	r.logger.LogCtx(ctx, "debug", "found the current version of the reconciled pod in the Kubernetes API")
+
+	// TODO drain guest cluster node and only remove the finalizer as soon as the
+	// guest cluster node associated with the reconciled host cluster pod is
+	// drained.
+
+	// TODO go ahead and do not block once the draining is initialized. Using
+	// finalizers on the pods the delete event will be replayed and we can check
+	// if the draining completed on the next reconciliation loop.
```

Comment:
All this is tested so far and works. The TODOs are going to be addressed in the next steps. Should be fairly simple to get the draining going via the node-operator.